File: SupplierPortal.Web/Modules/ServerTypes/Market/RequestColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CurrencyType } from "../Modules/Market.Request.Enum.CurrencyType";
import { RequestRow } from "./RequestRow";

export interface RequestColumns {
    Id: Column<RequestRow>;
    CompanyName: Column<RequestRow>;
    OfferId: Column<RequestRow>;
    OfferDate: Column<RequestRow>;
    ValueDate: Column<RequestRow>;
    RequestState: Column<RequestRow>;
    CurrencyType: Column<RequestRow>;
    Exchangerate: Column<RequestRow>;
    DeliveryTime: Column<RequestRow>;
    Representatives: Column<RequestRow>;
    ToWho: Column<RequestRow>;
    EMail: Column<RequestRow>;
    CallNumber: Column<RequestRow>;
}

export class RequestColumns extends ColumnsBase<RequestRow> {
    static readonly columnsKey = 'Market.Request';
    static readonly Fields = fieldsProxy<RequestColumns>();
}

[CurrencyType]; // referenced types

File: SupplierPortal.Web/Modules/Market/Request/RequestColumns.cs
using Serenity.ComponentModel;
using SupplierPortal.Market.Enum;
using System;
using System.ComponentModel;

namespace SupplierPortal.Market.Columns;

[ColumnsScript("Market.Request")]
[BasedOnRow(typeof(RequestRow), CheckNames = true)]
public class RequestColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string CompanyName { get; set; }
    public int RequestId { get; set; }
    public DateTime OfferDate { get; set; }
    public DateTime ValueDate { get; set; }
    public short State { get; set; }
    public short Currency { get; set; }
    public short CurrencyType { get; set; }
    public double Exchangerate { get; set; }
    public DateTime DeliveryTime { get; set; }
    [EditLink]
    public string Representatives { get; set; }
    public string ToWho { get; set; }
    public string EMail { get; set; }
    public string CallNumber { get; set; }
}